tests/dialects/test_x86.py

```diff
@@ -33,3 +33,45 @@ def test_rflags_register():
     rflags = x86.register.RFLAGS
     assert rflags.is_allocated
     assert rflags.register_name == "rflags"
+
+
+@pytest.mark.parametrize(
+    "register, name",
+    [
+        (x86.register.ZMM0, "zmm0"),
+        (x86.register.ZMM1, "zmm1"),
+        (x86.register.ZMM2, "zmm2"),
+        (x86.register.ZMM3, "zmm3"),
+        (x86.register.ZMM4, "zmm4"),
+        (x86.register.ZMM5, "zmm5"),
+        (x86.register.ZMM6, "zmm6"),
+        (x86.register.ZMM7, "zmm7"),
+        (x86.register.ZMM8, "zmm8"),
+        (x86.register.ZMM9, "zmm9"),
+        (x86.register.ZMM10, "zmm10"),
+        (x86.register.ZMM11, "zmm11"),
+        (x86.register.ZMM12, "zmm12"),
+        (x86.register.ZMM13, "zmm13"),
+        (x86.register.ZMM14, "zmm14"),
+        (x86.register.ZMM15, "zmm15"),
+        (x86.register.ZMM16, "zmm16"),
+        (x86.register.ZMM17, "zmm17"),
+        (x86.register.ZMM18, "zmm18"),
+        (x86.register.ZMM19, "zmm19"),
+        (x86.register.ZMM20, "zmm20"),
+        (x86.register.ZMM21, "zmm21"),
+        (x86.register.ZMM22, "zmm22"),
+        (x86.register.ZMM23, "zmm23"),
+        (x86.register.ZMM24, "zmm24"),
+        (x86.register.ZMM25, "zmm25"),
+        (x86.register.ZMM26, "zmm26"),
+        (x86.register.ZMM27, "zmm27"),
+        (x86.register.ZMM28, "zmm28"),
+        (x86.register.ZMM29, "zmm29"),
+        (x86.register.ZMM30, "zmm30"),
+        (x86.register.ZMM31, "zmm31"),
+    ],
+)
+def test_avx_register(register: x86.register.AVXRegisterType, name: str):
```

Comment:
I'm not sure we need all these additional tests? I feel that they bloat test time which is already getting quite large imo

Reply:
@webmiche suggested doing a parametrised test like that for general purposes registers (https://github.com/xdslproject/xdsl/pull/2333#discussion_r1526190469) so I wanted to keep it consistent.